File: 백준/Silver/1065. 한수/한수.cc
#include <iostream>

using namespace std;

int arithmetic(int num) {
    int count = 0;
    
    if(num < 100) {
        count = num;
    } else {
        count = 99;
        
        for(int i = 100; i<num+1; i++) {
            int hun = i / 100;
            int ten = (i / 10) % 10;
            int one = i % 10;
        
            if((hun - ten) == (ten - one)) {
                count ++;
            } else {
                // empty
            }
        }
    }
    
    return count;
}

int main() {
    int num = 0;
    
    cin >> num;
    
    cout << arithmetic(num) << endl;
    
    return 0;
}

File: 백준/Silver/1065. 한수/한수.swift
import Foundation

let N = Int(readLine()!)!
var count = 99

if N<100{
    print(N)
}
else if N==1000{
    print(144)
}
else{
    for i in 100...N{
        let n1 = (i%1000)/100
        let n2 = (i%100)/10
        let n3 = (i%10)/1
        
        if (n2-n1) == (n3-n2){
            count += 1
        }
    }
    print(count)